python/cudf/cudf/_libxx/sort.pyx

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+from __future__ import print_function
+import pandas as pd
+
+from cudf._libxx.column cimport *
+from cudf._libxx.table cimport *
+from cudf._libxx.lib cimport *
+
+from cudf._libxx.includes.sort cimport *
+
+
+def order_by(Table source_table, ascending, na_position):
+    """
+    Sorting the table ascending/descending
+
+    Parameters
+    ----------
+    source_table : table which will be sorted
+    ascending : list of boolean values which correspond to each column
+                in source_table signifying order of each column
+                True - Ascending and False - Descending
+    na_position : whether null should be considered larget or smallest value
+                  0 - largest and 1 - smallest
+
+    """
+
+    cdef table_view source_table_view = source_table.data_view()
+    cdef vector[order] column_order
+    cdef vector[null_order] null_precedence
+
+    for i in ascending:
+        if i is True:
+            column_order.push_back(order.ASCENDING)
+        else:
+            column_order.push_back(order.DESCENDING)
+
+    cdef null_order pred = (
+        null_order.BEFORE if na_position == 1 else null_order.AFTER)
+
+    for i in range(source_table._num_columns):
+        null_precedence.push_back(pred)
+
+    cdef unique_ptr[column] c_result
+    with nogil:
+        c_result = move(sorted_order(source_table_view,
+                                     column_order,
+                                     null_precedence))
+
+    return Column.from_unique_ptr(move(c_result))
+
+
+def digitize(Table source_values_table, Table bins, right=False):
+    """
+    Return the indices of the bins to which each value in source_table belongs.
+
+    Parameters
+    ----------
+    source_table : Input table to be binned.
+    bins : Table containing columns of bins
+    right : Indicating whether the intervals include the
+            right or the left bin edge.
+    """
+
+    cdef table_view bins_view = bins.view()
+    cdef table_view source_values_table_view = source_values_table.view()
+    cdef vector[order] column_order
+    cdef vector[null_order] null_precedence
+    for i in range(bins_view.num_columns()):
+        column_order.push_back(order.ASCENDING)
+        null_precedence.push_back(null_order.BEFORE)
```

Comment:
Instead of looping and using `push_back` here you should be able to just default initialize the vectors above:
```
cdef vector[order] column_order = vector[order](bins_view.num_columns(), order.ASCENDING)
cdef vector[null_order] null_precedence = vector[null_order](bins_view.num_columns(), null_order.BEFORE)
```